### ISSUE TITLE
Clarify documentation for fig.add_subplot()

### ISSUE THREAD

**Issue by butlerbj** [matplotlib/matplotlib#10358]
New to matplotlib and python in general, so my apologies if this is just my own ignorance. 

The [doc](https://matplotlib.org/api/_as_gen/matplotlib.figure.Figure.html#matplotlib.figure.Figure.add_subplot) for `add_subplot` says:

> *args
Either a 3-digit integer or three separate integers describing the position of the subplot. If the three integers are I, J, and K, the subplot is the Ith plot on a grid with J rows and K columns.

The doc implies that the three integers are appearing in that order, and I understood this to mean:
`fig.add_subplot(I, J, K)`, 
when in in practice, the syntax (according to the current documentation) should be:
`fig.add_subplot(J, K, I)` 
because the last of the three integers represents the order/placement of the plot, and the first two represent the rows and columns.

I think it would be clearer if the doc read as:

> *args
Either a 3-digit integer or three separate integers describing the position of the subplot. If the three integers are I, J, and K (in that order) the subplot is the _Kth_ plot on a grid with I rows and J columns. 

Further, I think the doc could benefit from an example that directly shows adding multiple subplots. I can make this contribution myself, but would need to be pointed in the right direction!

**Comment by anntzer** [comment]
Agree with the suggested fix.  We may also want to harminize the description with the one in `pyplot.subplot`.

See https://matplotlib.org/devdocs/devel/documenting_mpl.html (https://matplotlib.org/devdocs/devel/index.html if you are courageous enough to read the whole thing :-)) for some help on writing documentation, or just try making a PR and we'll help you get it in :-)

**Comment by timhoffm** [comment]
Adding call signatures would also be helpful:
~~~
add_subplot(rows, cols, index, projection=None, polar=False, **kwargs)
add_subplot(pos, projection=None, polar=False, **kwargs)
~~~
cf. [subplot](https://matplotlib.org/api/_as_gen/matplotlib.pyplot.subplot.html#matplotlib.pyplot.subplot)

**Comment by timhoffm** [comment]
#10394 fixes the order issue. The other suggestions should still be addressed.

**Comment by dstansby** [comment]
Fixed by #10394
